client/scripts/views/pages/view_proposal/index.ts

```diff
@@ -73,22 +76,21 @@ const ProposalHeader: m.Component<{
       chain: app.activeChainId(),
       community: app.activeCommunityId()
     }));
-    const canEdit = app.user.activeAccount?.address === proposal.author
-      && app.user.activeAccount?.chain.id === (proposal as OffchainThread).authorChain;
+
+    // Original posters have full editorial control, while added collaborators
+    // merely have access to the body and title
+    vnode.state.isAuthor = (app.user.activeAccount?.address === proposal.author
+          && app.user.activeAccount?.chain.id === (proposal as OffchainThread).authorChain);
+    vnode.state.isEditor = (proposal as OffchainThread).collaborators?.filter((c) => {
+      return (c.address === app.user.activeAccount?.address && c.chain === app.user.activeAccount?.chain.id);
+    }).length > 0;
+
+    const { isAuthor, isEditor, isAdmin } = vnode.state;
```

Comment:
(as above) This seems unintended, why are we assigning to vnode.state every time the view renders?